### ISSUE TITLE
Version .7 won't even load..Stuck on "starting excavator server"

### ISSUE THREAD

**Issue by yearight** [nicehash/NiceHashMiner2#79]
I do have the 2013 redistributable installed.

**Comment by Crypto-Lord** [comment]
It may take a few minutes to load, I was noticing it was taking a long time to getting past the 'Starting Excavator Server'

**Comment by S74nk0** [comment]
**excavator** was updated to [v1.2.6a](https://github.com/nicehash/excavator/releases/tag/v1.2.6a). 
@yearight please send the logs for v2.0.0.7 to see what is causing issues.

**Comment by S74nk0** [comment]
@yearight if you can please try running [v1.2.6a](https://github.com/nicehash/excavator/releases/tag/v1.2.6a) standalone with '-p 4000 -f 0'. The issue is most likely related with excavator.

**Comment by Crypto-Lord** [comment]
@S74nk0 for Excavator Server to work properly it requires disabling driver signature enforcement. I manage to do that but any way of putting a note for others who are unsure on how to do this?

**Comment by S74nk0** [comment]
@Crypto-Lord to achieve the full speed on AMD (future algos to come to nhm2) and install **phymem.sys** yes. But it shouldn't crash because of it.

**Comment by yearight** [comment]
I did let it try to load for a bit...never worked for me, thanks though.
[log.txt](https://github.com/nicehash/NiceHashMiner2/files/1079219/log.txt)

If I run the bat file for excavator the webpage does open, but is currently offline. 

Same thing with the updated excavator. Interestingly, when I tried to paste it, excavator was still running...I had to use task manager to shut it down.
New excavator log... 
[log.txt](https://github.com/nicehash/NiceHashMiner2/files/1079261/log.txt)

Crypto-Lord....does that mean they simply have to sign their driver and problem solved? Or is it someone elses...else(is elses a word lol).... driver? Wonder if someone forgot to sign it.

**Comment by S74nk0** [comment]
@yearight Did the previous NHM2 versions (5 or 6) block at startup?

**Comment by yearight** [comment]
No...6 worked for me, but was way less profitable than Version 1.7.5.13 so I never ran it long.

**Comment by Crypto-Lord** [comment]
@yearight if you don't disable driver signature windows will block 'phymem.sys' from installing. There's 2 ways off disabling it (1 Temporarily, the other permanently until re-enabled) Link below with both options for disabling it.

http://windowsreport.com/driver-signature-enforcement-windows-10/ 

Basically Windows requires digitally signed drivers to prevent unwanted malware from sketchy sources. Like @S74nk0 said above to achieve full AMD speed this will need to be disabled.

**Comment by yearight** [comment]
I guess the new excavator includes that 'phymem.sys' file, because its in my folder, not sure if it was before. Copy and pasting maybe doesn't work, maybe it needs registering or something...it is a driver. I thought drivers are in the windows folder?

That would explain why the new version is so much less profitable too.

I still wonder if they could just sign the driver...but I'd imagine theres verification and other things that take time.

"@yearight if you can please try running v1.2.6a standalone with '-p 4000 -f 0''....I missed the "-p 400 -f 0" part earlier...was I suppose to input that somewhere?

**Comment by Squall-Leonhart** [comment]
windows 7 and 8 allow third party signing.
windows 10 only allows signing by microsoft hardware labs.

probably shouldn't be trying (and failing) to load AMD crab, phymem or whatever on systems that don't have any amd parts in them.

**Comment by S74nk0** [comment]
@everyone excavator has been updated. Please restart **nhm2007** (it will update excavator) and report if it finishes loading.

**Comment by yearight** [comment]
Still doesn't load...no change.

**Comment by S74nk0** [comment]
@yearight When this happens is **excavator** active in Task Manager?

**Comment by yearight** [comment]
Yes...and it is still there when I close the program..I have to end task.

**Comment by yearight** [comment]
But I get no BSOD.
Disabling driver signature and installing changes nothing for me unfortunately.

**Comment by stillagainst** [comment]
Here is what I've been through:

First, from an administrative promt I've executed: bcdedit /set testsigning on

This turned Windows 10 in a Test Mode so driver signature enforcement is turned off.
This is good for machines where you dont have physical access right now.
(find out more about test mode [here](https://www.howtogeek.com/167723/how-to-disable-driver-signature-verification-on-64-bit-windows-8.1-so-that-you-can-install-unsigned-drivers/) )

But still no success.
So it's been said that the issue is that phymem.sys is not yet digitally signed so excavator doesn't start.

It is located there: %AppData%\nhm2\bin\excavator_server
Right clicked on phymem.sys, digital signatures, it says there are valid signatures from Zawawa and Microsoft since June 17, seems legit.

So I've tried to run excavator.exe from elevated promt, and it starts fine.
But if I try to stop it via Ctrl+C, it says it will shutdown, but it hangs there and shows load in Task Manager, where I have to kill it manually.

Furthermore there is a batch file named: excavator+web+restart_script.bat
This one runs excavator, runs a test bench and a website displaying some stats.
That works well for me, too.
But then again, I close the page, try to stop that script, nothing happens, have to kill it the hard way.

To sum up:
I assume when NHM2.0.0.7 starts, it tries to test if excavator is working, it starts, but NHM can't stop the test, so it hangs there and still says "starting" (and confuses us).
Again I can kill excavator in Task Manager and NHM will finish loading, but can't run half of the miners and makes less provit than 1.x

**Comment by S74nk0** [comment]
**everyone** we have released a new [excavator_v1.2.9a](https://github.com/nicehash/excavator/releases/tag/v1.2.9a) and it should have this issue fixed. After NiceHash Miner 2.0.0.7 restart it will update excavator to **v1.2.9a**.
Please report if **NiceHash Miner 2.0.0.7** and **excavator v1.2.9a** resolve this issue.

**Comment by yearight** [comment]
v1.2.9a makes no difference for me.

**Comment by ghost** [comment]
BSOD on excavator server launch.

**Comment by stillagainst** [comment]
Update launched, no BSOD for me and none before, same result as I described above.

Manually launched excavator outputs:

=========================== www.nicehash.com =========================
                Excavator v1.2.9a GPU Miner for NiceHash.
           Copyright (C) 2017 NiceHash. All rights reserved.
Developed by djeZo, zawawa and dropky with help and contributions from
             pallas, Vorksholk, bitbandi, ocminer, Genoil
=========================== www.nicehash.com =========================

Build time: 2017-06-29 16:56:02
Build number: 512
Provided startup commandline:
        "C:\Users\bob\AppData\Roaming\nhm2\bin\excavator_server\excavator.exe"

[11:53:32][0x00000dd0][info] Log started
[11:53:33][0x00000dd0][info] opencl | Loaded phymem for direct PCIe access.
[11:53:33][0x00000dd0][info] core | Found OpenCL device: AMD Radeon R9 200 Series
[11:53:33][0x00000dd0][info] opencl |     PCI Vendor/Device ID: 0x1002, 0x6810
[11:53:33][0x00000dd0][info] opencl |     PCI Subsystem Vendor/Device ID: 0x174b, 0xe271
[11:53:33][0x00000dd0][info] core | Found OpenCL device: AMD Radeon HD 7700 Series
[11:53:33][0x00000dd0][info] opencl |     PCI Vendor/Device ID: 0x1002, 0x683f
[11:53:33][0x00000dd0][info] opencl |     PCI Subsystem Vendor/Device ID: 0x1043, 0x0427
[11:53:33][0x00000dd0][info] api | Listening on 127.0.0.1:3456
[11:53:33][0x00000dd0][info] core | Initialized!

And after Ctrl+C:

[11:53:52][0x00000dd0][info] Shutting down
[11:53:52][0x00000dd0][info] api | Closing

And it hangs there.

**Comment by yearight** [comment]
It does load for me now...just takes a long time. Not sure how long it takes, but could be 4 or 5 minutes.
I might not have left it long enough with previous versions so I don't know which version fixed it.
The excavator cmd window stays open, is this normal?

**Comment by stillagainst** [comment]
Tried it again, works for me now aswell.

Yes excavator window staying up is normal, afair it launches this server on localhost port 4000 and the ethminer window connects to it and through that it gets direct PCIe access with phymem.

**Comment by S74nk0** [comment]
NiceHash Miner **v2.0.0.8 Alpha** is released and includes **feature**/**bugfix** related to this issue. Expect an update notice in your NHM2 client or download the new version manually.
Closing...
Reopen if issue is present in **v2.0.0.8 Alpha**

**Comment by rosdi** [comment]
Hi @S74nk0 , I am using NoceHash Miner 2.0.1.1 - Beta, so far I have wait for more than 20 minutes but it is still stuck at 'Starting Excavator Server...'

![image](https://user-images.githubusercontent.com/5094028/30768516-fe5de45a-a03a-11e7-8ff3-2fd6373b5288.png)

**Comment by rosdi** [comment]
Well... it finally loads now... not sure how long I waited... I went for breakfast and when I came back it has finished loading...

-- UPDATE --
I have timed it, it took **34 minutes** to load...

**Comment by xwinata** [comment]
^ i got same issue, mine never got started. it became like this in midnight and just realized it in the morning. i have to restart the NM to get it running again.
it already happened 3-4 times since i installed 2.0.1.1 NM

**Comment by ronaldobini** [comment]
I have the same issue in v2.0.1.5. 

UPDATE>  the time/day of the computer was worng, after fixing the date on the system, everything ran ok

**Comment by jagauthier** [comment]
I found this thread after searching... yes, my NMH does the same thing.    I did not wait 35 minutes to see.. more like 10.  It looks like my excavator service isn't running in Task Manager, so I can't be sure it even downloads.  I can't seem to find a binary for it anywhere.
